geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/pubsub/PubSubExecutor.java

```diff
@@ -33,28 +33,57 @@
   @Override
   public RedisResponse executeCommand(Command command, ExecutionHandlerContext context) {
 
-    byte[] subCommand = command.getProcessedCommand().get(1);
+    List<byte[]> processedCommand = command.getProcessedCommand();
+    String subCommand = Coder.bytesToString(processedCommand.get(1)).toLowerCase();
 
-    if (!Arrays.equals(subCommand, Coder.stringToBytes("channels"))) {
-      return RedisResponse
-          .error(String.format(ERROR_UNKNOWN_PUBSUB_SUBCOMMAND, new String(subCommand)));
-    }
+    List<Object> response;
+
+    switch (subCommand) {
+      case "channels":
+        // in a subsequent story, a new parameter requirement class
+        // specific to subCommands might be a better way to do this
+        if (processedCommand.size() > 3) {
+          return RedisResponse
+              .error(String.format(ERROR_UNKNOWN_PUBSUB_SUBCOMMAND, subCommand));
+        }
+
+        response = doChannels(processedCommand, context);
+        break;
+
+      case "numsub":
+        response = doNumsub(processedCommand, context);
+        break;
+
+      default:
+        return RedisResponse
+            .error(String.format(ERROR_UNKNOWN_PUBSUB_SUBCOMMAND, subCommand));
 
-    // in a subsequent story, a new parameter requirement class
-    // specific to subCommands might be a better way to do this
-    if (command.getProcessedCommand().size() > 3) {
-      return RedisResponse
-          .error(String.format(ERROR_UNKNOWN_PUBSUB_SUBCOMMAND, new String(subCommand)));
     }
 
-    List<byte[]> response;
-    if (command.getProcessedCommand().size() > 2) {
-      byte[] pattern = command.getProcessedCommand().get(2);
+    return RedisResponse.array(response);
```

Comment:
To avoid having to change the return type of the `findChannelNames()` methods, you could return directly from each switch case, which would mean that you wouldn't need to define `List<Object> response` above.

Reply:
yeah, that's a bit better-  thanks